src/test/mir-opt/simplify-arm.rs

```diff
@@ -20,8 +20,23 @@ fn id_result(r: Result<u8, i32>) -> Result<u8, i32> {
     }
 }
 
+fn into_result<T, E>(r: Result<T, E>) -> Result<T, E> {
+    r
+}
+
+fn from_error<T, E>(e: E) -> Result<T, E> {
+    Err(e)
+}
+
+// This was written to the `?` from `try_trait`,
+// but `try_trait_v2` uses a different structure,
+// so the relevant desugar is copied inline
+// in order to keep the test testing the same thing.
 fn id_try(r: Result<u8, i32>) -> Result<u8, i32> {
```

Comment:
same as above, if you don't want to address this in this PR, please add a fix me.

I don't think it makes much sense to test how the old try impl desugars, considering that this code won't be written manually.